substrate/frame/delegated-staking/src/lib.rs

```diff
@@ -0,0 +1,820 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![deny(rustdoc::broken_intra_doc_links)]
+
+//! An implementation of a delegation system for staking that can be utilised using
+//! [`DelegationInterface`]. In future, if exposed via extrinsic, these primitives could also be
+//! used by off-chain entities, smart contracts or by other parachains via xcm.
+//!
+//! Delegatee: Someone who accepts delegations. An account can set their intention to accept
```

Comment:
Just calling the delegatee the "delegate" sounds a bit more natural to me. Delegators and delegate.

Delegate - a person or entity sent or authorized to represent others.
Delegator - someone who delegates (= gives) part of their job, duties, etc. to someone else to do for them.